packages/mui-material/src/Tabs/Tabs.js

```diff
@@ -667,7 +651,7 @@ const Tabs = React.forwardRef(function Tabs(inProps, ref) {
       firstObserver?.disconnect();
       lastObserver?.disconnect();
     };
-  }, [scrollable, scrollButtons, childrenProp?.length]);
+  }, [scrollable, scrollButtons, updateScrollObserver, childrenProp?.length]);
```

Comment:
Shouldn't this be checked on the if condition above to run once it's `true`?

Reply:
No, it should run after changing one of the three props and also each time the developer calls the action updateScrollButtons